sympy/solvers/inequalities.py

```diff
@@ -1007,3 +1009,154 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+class UnboundedLinearProgrammingError(Exception):
+    pass
+
+
+class InfeasibleLinearProgrammingError(Exception):
+    pass
+
+
+def _pivot(M, i, j):
+    MM = Matrix.zeros(M.rows, M.cols)
+    for ii in range(M.rows):
+        for jj in range(M.cols):
+            if ii == i and jj == j:
+                MM[ii, jj] = 1 / M[i, j]
+            elif ii == i and jj != j:
+                MM[ii, jj] = M[ii, jj] / M[i, j]
+            elif ii != i and jj == j:
+                MM[ii, jj] = -M[ii, jj] / M[i, j]
+            else:
+                MM[ii, jj] = M[ii, jj] - M[ii, j] * M[i, jj] / M[i, j]
```

Comment:
This can be more vectorised. Something like:
```python
def _pivot2(M, i, j):
    Mi, Mj, Mij = M[i,:], M[:,j], M[i,j]
    MM = M - Mj * (Mi / Mij)
    MM[i,:] = Mi / Mij
    MM[:,j] = -Mj / Mij
    MM[i,j] = 1 / Mij
    return MM
```
For large matrices this is faster e.g.
```python
In [25]: M = randMatrix(100)

In [26]: %time ok = _pivot(M, 2, 2)
CPU times: user 696 ms, sys: 5.19 ms, total: 701 ms
Wall time: 704 ms

In [27]: %time ok = _pivot2(M, 2, 2)
CPU times: user 93.6 ms, sys: 3.81 ms, total: 97.4 ms
Wall time: 98.8 ms
```